backend/pystac_io_methods.py

```diff
@@ -0,0 +1,28 @@
+from typing import IO, Any, Union
+from urllib.parse import urlparse
+
+import boto3
+from pystac import STAC_IO  # type: ignore[import]
+
+
+def read_method(uri: str) -> Any:
+    parsed = urlparse(uri)
+    if parsed.scheme == "s3":
+        bucket = parsed.netloc
+        key = parsed.path[1:]
+        s3_resource = boto3.resource("s3")
```

Comment:
Is this a resource we could instantiate outside of the function to make sure it's initialized at import time?